rds/rds_provider.go

```diff
@@ -152,40 +157,31 @@ func getEndpoint(dbName *string, svc *rds.Client) (string, error) {
 	return dbHostname, nil
 }
 
-func (r *RDS) DeleteDatabase(db *crd.Database) error {
+func (r *RDS) DeleteDatabase(ctx context.Context, db *crd.Database) error {
 	if db.Spec.DeleteProtection {
 		log.Printf("Trying to delete a %v in %v which is a deleted protected database", db.Name, db.Namespace)
 		return nil
 	}
 	// delete the database instance
 	svc := r.rdsclient()
-	res := svc.DeleteDBInstanceRequest(&rds.DeleteDBInstanceInput{
+
+	_, err := svc.DeleteDBInstance(ctx, &rds.DeleteDBInstanceInput{
 		DBInstanceIdentifier: aws.String(dbidentifier(db)),
-		SkipFinalSnapshot:    aws.Bool(true),
+		SkipFinalSnapshot:    true,
 	})
-	_, err := res.Send(context.Background())
-	if err != nil {
-		e := errors.Wrap(err, fmt.Sprintf("unable to delete database %v", db.Spec.DBName))
-		log.Println(e)
-		return e
-	} else {
-		log.Printf("Waiting for db instance %v to be deleted\n", db.Spec.DBName)
-		time.Sleep(5 * time.Second)
 
-		k := &rds.DescribeDBInstancesInput{DBInstanceIdentifier: aws.String(dbidentifier(db))}
-		err = r.rdsclient().WaitUntilDBInstanceDeleted(context.Background(), k)
```

Comment:
Why remove this?

Reply:
when library `aws-go-sdk-v2` upgrade from v0.11.0 to v1.3.0, this `WaitUntilDBInstanceDeleted` api was removed from the library.

Reply:
> when library `aws-go-sdk-v2` upgrade from v0.11.0 to v1.3.0, this `WaitUntilDBInstanceDeleted` api was removed from the library.

thanks for joining in @wgarunap

Reply:
It seems they move to https://aws.github.io/aws-sdk-go-v2/docs/making-requests/#using-waiters I assume there is something for RDS
I think we should use that so we don't change the semantic of the API.

Reply:
Can't find a waiter for rds in the entirety of [https://www.github.com/aws/aws-sdk-go-v2/tree/main/service%2Frds](the git module) will check if it's implemented else where.

Reply:
Found waiters for other operations as you described, but didn't see one for the delete operations.

https://github.com/aws/aws-sdk-go-v2/blob/service/rds/v1.2.1/service/rds/api_op_DeleteDBCluster.go

https://github.com/aws/aws-sdk-go-v2/blob/service/rds/v1.2.1/service/rds/api_op_DeleteDBInstance.go